### PR TITLE
Add an msbuild property to pass a JavaScript engine in WASM test build

### PR DESCRIPTION
**Pull request by MaximLipnin**
It adds the use of `/p:JSEngine` property, which is, by default, `V8`.

### CODE REVIEW

File: eng/testing/WasmRunnerTemplate.sh [hunk 1]
```diff
@@ -1,17 +1,16 @@
 #!/usr/bin/env bash
 
-EXECUTION_DIR=$(dirname $0)
-
-echo "Test: $1"
+JAVASCRIPT_ENGINE=$0
```
Comment (akoeplinger) [inline eng/testing/WasmRunnerTemplate.sh:3]: `$0` is the name of the script so this needs to stay the way it was before and you need to make `JAVASCRIPT_ENGINE=$1` instead.

**Review by akoeplinger — CHANGES_REQUESTED** [review]

File: src/mono/wasm/Makefile [hunk 2]
```diff
@@ -118,5 +118,9 @@ test-runner:
...
 	$(DOTNET) build $(TOP)/tools-local/tasks/mobile.tasks/WasmAppBuilder
 
-run-tests-%:
-	PATH="$(JSVU):$(PATH)" $(DOTNET) build $(TOP)/src/libraries/$*/tests/ /t:Test /p:TargetOS=Browser /p:TargetArchitecture=wasm /p:Configuration=$(CONFIG)
+run-tests-%-v8:
+	PATH="$(JSVU):$(PATH)" $(DOTNET) build $(TOP)/src/libraries/$*/tests/ /t:Test /p:TargetOS=Browser /p:TargetArchitecture=wasm /p:Configuration=$(CONFIG) /p:JSEngine=V8
+run-tests-%-sm:
+	PATH="$(JSVU):$(PATH)" $(DOTNET) build $(TOP)/src/libraries/$*/tests/ /t:Test /p:TargetOS=Browser /p:TargetArchitecture=wasm /p:Configuration=$(CONFIG) /p:JSEngine=SpiderMonkey
+run-tests-%-jsc:
+	PATH="$(JSVU):$(PATH)" $(DOTNET) build $(TOP)/src/libraries/$*/tests/ /t:Test /p:TargetOS=Browser /p:TargetArchitecture=wasm /p:Configuration=$(CONFIG) /p:JSEngine=JavaScriptCore
```
[the remark this answers is not in the document]
Comment (MaximLipnin) [reply src/mono/wasm/Makefile]: `run-tests-%: run-tests-v8-%` doesn't work

File: eng/testing/WasmRunnerTemplate.sh (hunk 1 above)
Comment (MaximLipnin) [reply eng/testing/WasmRunnerTemplate.sh]: done

File: eng/testing/tests.targets [hunk 3]
```diff
@@ -83,7 +83,7 @@
       <RunTestsCommand>"$(RunScriptOutputPath)" --runtime-path "$(TestHostRootPath.TrimEnd('\/'))"</RunTestsCommand>
       <RunTestsCommand Condition="'$(TestRspFile)' != '' and '$(RuntimeFlavor)' != 'Mono'">$(RunTestsCommand) --rsp-file "$(TestRspFile)"</RunTestsCommand>
       <RunTestsCommand Condition="'$(TargetsMobile)' == 'true'">"$(RunScriptOutputPath)" $(AssemblyName) $(TargetArchitecture)</RunTestsCommand>
-      <RunTestsCommand Condition="'$(TargetOS)' == 'Browser'">"$(RunScriptOutputPath)" $(AssemblyName).dll $(_withoutCategories.Replace(';', ' -notrait category='))</RunTestsCommand>
+      <RunTestsCommand Condition="'$(TargetOS)' == 'Browser'">$(JSEngine) "$(RunScriptOutputPath)" $(AssemblyName).dll $(_withoutCategories.Replace(';', ' -notrait category='))</RunTestsCommand>
```
[the remark this answers is not in the document]
Comment (MaximLipnin) [reply eng/testing/tests.targets]: done

File: eng/testing/tests.mobile.targets [hunk 4]
```diff
@@ -4,6 +4,7 @@
     <BundleDir>$([MSBuild]::NormalizeDirectory('$(OutDir)', 'AppBundle'))</BundleDir>
     <RunScriptOutputPath>$([MSBuild]::NormalizePath('$(BundleDir)', '$(RunScriptOutputName)'))</RunScriptOutputPath>
     <RunAOTCompilation Condition="'$(TargetOS)' == 'iOS' and $(TargetArchitecture.StartsWith('arm'))">true</RunAOTCompilation>
+    <JSEngine Condition="'$(TargetOS)' == 'Browser' and $(JSEngine) ==''">V8</JSEngine>
```
Comment (akoeplinger) [inline eng/testing/tests.mobile.targets:7]:
```suggestion
    <JSEngine Condition="'$(TargetOS)' == 'Browser' and '$(JSEngine)' == ''">V8</JSEngine>
```

**Review by akoeplinger — APPROVED** [review]
one suggestion, looks good otherwise

**Review by steveisok — APPROVED** [review]

**Pull request merged**